src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/XamarinProject.cs

```diff
@@ -354,5 +358,50 @@ public virtual string ProcessSourceTemplate (string source)
 		{
 			return source.Replace ("${ROOT_NAMESPACE}", RootNamespace ?? ProjectName).Replace ("${PROJECT_NAME}", ProjectName);
 		}
+
+		public void CopyNuGetConfig (string relativeDirectory)
+		{
+			// Copy our solution's NuGet.config
+			var nuget_config = Path.Combine (XABuildPaths.TopDirectory, "NuGet.config");
+			if (File.Exists (nuget_config)) {
+				string projNugetConfig = Path.Combine (Root, relativeDirectory, "NuGet.config");
+				File.Copy (nuget_config, projNugetConfig, overwrite: true);
+				// Write additional sources to NuGet.config if needed
+				if (ExtraNuGetConfigSources != null) {
```

Comment:
nice :) I could have used this in #3938

Reply:
Feel free to steal it for that PR, I can rebase this if that PR ends up landing first or vice versa.